From: Ingo Schwarze <ischwarze@astaro.com>
Description: bugfix: cache content leaks to global
 In the read_cache sub, a hash is passed to $config for filling in from the
 cache file. However, read_cache writes to $config{...} instead of
 $config->{...}. Unfortunately, the latter overwrites the global config.
 This includes the new IP address passed by the user...
Bug-Debian: http://bugs.debian.org/527848
Reviewed-by: Torsten Landschoff <torsten@debian.org>

Index: b/ddclient
===================================================================
--- a/ddclient
+++ b/ddclient
@@ -846,9 +846,9 @@
 	%opt   = %saved;
 
 	foreach my $h (keys %cache) {
-	    if (exists $config{$h}) {
+	    if (exists $config->{$h}) {
 		foreach (qw(atime mtime wtime ip status)) {
-	    	    $config{$h}{$_} = $cache{$h}{$_} if exists $cache{$h}{$_};
+	    	    $config->{$h}{$_} = $cache{$h}{$_} if exists $cache{$h}{$_};
 		}
 	    }
 	}
@@ -1098,7 +1098,7 @@
 	## merge options into host definitions or globals
 	if (@hosts) {
 	    foreach my $h (@hosts) {
-		$config{$h} = %{ merge(\%options, $config{$h}) };
+		$config{$h} = merge(\%options, $config{$h});
 	    }
 	    $opt{'host'} = join(',', @hosts);
 	} else {
